### PR TITLE
Update repo name and package after migration.

### PR DESCRIPTION
**Pull request by mneverov**

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -11,9 +11,9 @@ on:
 env:
   K8S_VERSION: "v1.27.3"
   KIND_VERSION: "v0.20.0"
-  KIND_CLUSTER_NAME: "clustercidr"
+  KIND_CLUSTER_NAME: "nodeipam"
   REGISTRY: ghcr.io
-  IMAGE_NAME: mneverov/cluster-cidr-controller
+  IMAGE_NAME: sigs.k8s.io/node-ipam-controller
```
Comment (ameukam) [inline .github/workflows/e2e.yml:16]: This is invalid. there is no registry `sigs.k8s.io` unless you want to do a local build ?

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -30,14 +30,14 @@ jobs:
 
     - name: Build docker
       run: |
-        docker build -t ghcr.io/mneverov/cluster-cidr-controller:ci-e2e-test -f Dockerfile .
+        docker build -t ghcr.io/sigs.k8s.io/node-ipam-controller:ci-e2e-test -f Dockerfile .
```
Comment (ameukam) [inline .github/workflows/e2e.yml:33]: Can we use GCP GCR instead even local/step builds ? GHCR is currently not enabled

File: charts/node-ipam-controller/values.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 replicaCount: 1
 
 image:
-  repository: ghcr.io/mneverov/cluster-cidr-controller
+  repository: ghcr.io/sigs.k8s.io/node-ipam-controller
```
Comment (ameukam) [inline charts/node-ipam-controller/values.yaml:4]: A TODO that we want to use a different repo?

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (aojea) [reply .github/workflows/e2e.yml:33]: I think you need to give some directions Arnaud on how to do it

Comment (mneverov) [reply .github/workflows/e2e.yml:33]: my bad. I understand that it is possible to use https://github.com/kubernetes/registry.k8s.io, right?
Also, my understanding that it is not `ghcr.io/sigs.k8s.io/node-ipam-controller` but `ghcr.io/kubernetes-sigs/node-ipam-controller` and it should work, correct?

Comment (ameukam) [reply .github/workflows/e2e.yml:33]: Should be `gcr.io/k8s-staging-networking/node-ipam-controller`. 
Also push the image to a registry can't be currently done with Github Actions.

Comment (mneverov) [reply .github/workflows/e2e.yml:33]: @ameukam my understanding is that this [PR](https://github.com/kubernetes/test-infra/pull/31636) is a prerequisite to this, right?

**Pull request merged**